rclcpp/test/benchmark/benchmark_parameter.cpp

```diff
@@ -0,0 +1,225 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "performance_test_fixture/performance_test_fixture.hpp"
+
+#include "rclcpp/rclcpp.hpp"
+
+class ParameterTest : public performance_test_fixture::PerformanceTest
+{
+public:
+  ParameterTest()
+  : node_name("my_node"),
+    param_prefix("my_prefix"),
+    param1_name(param_prefix + ".my_param_1"),
+    param2_name(param_prefix + ".my_param_2"),
+    param3_name(param_prefix + ".my_param_3"),
+    param_value1(true),
```

Comment:
What do you think about testing an array type parameter, and possible a string parameter (since it's not just a pod). I don't think you need to test all the API against all types of parameters, but maybe having declare with different types, and then having set/list with many different types?

Reply:
I added a basic array and string test and named the existing type-specific tests to reflect that they are using a bool.